### PR TITLE
Set topic for voice channels

### LINKED ISSUE #77: Set topic for voice channels
`s!setvoicetopic` `s!svt` renames the current voice channel that the user is in into a topic. When there is no longer anybody in the voice channel, the channel's name is reverted.

### PR DESCRIPTION
**Pull request by Programmerryoki**
Resolves #77 

Notes : 
 - The Voice Channel can only be renamed every 10minutes.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,19 +1,28 @@
 const Discord = require('discord.js');
 const { prefix, API_TOKEN } = require('./config.json');
 const { CommandHandler } = require('./src/CommandHandler');
+const { VoiceChannelChangeHandler } =
```
Comment (jerrylee17) [inline server.js:4]: For lines 4 and 5, change it to this:
```
const {
  VoiceChannelChangeHandler
} = require('./src/VoiceChannelChangeHandler');
```

This allows for better readability

File: src/CommandHandler.js [hunk 2]
```diff
@@ -65,7 +66,12 @@ class CommandHandler {
     const commandName = args.shift().toLowerCase();
     args = parseCommandParameters(args.join(' '));
 
+    // if (commandName === "ccc" || commandName === 'clearcooldown')
```
Comment (jerrylee17) [inline src/CommandHandler.js:69]: Remove comments

File: src/CommandHandler.js [hunk 3]
```diff
@@ -36,6 +36,7 @@ class CommandHandler {
     for (const directory in commandFiles) {
       for (const file in commandFiles[directory]) {
         const command = require(`${this.commandPath}/${directory}/${file}`);
+        // console.log(command);
```
Comment (jerrylee17) [inline src/CommandHandler.js:39]: Remove comments

File: src/CommandHandler.js [hunk 4]
```diff
@@ -65,7 +66,12 @@ class CommandHandler {
     const commandName = args.shift().toLowerCase();
     args = parseCommandParameters(args.join(' '));
 
+    // if (commandName === "ccc" || commandName === 'clearcooldown')
+    // {
+    //   this.cooldownManager.
+    // } else 
     if (!this.commandMap.has(commandName)) {
+      console.log('no such command');
```
Comment (jerrylee17) [inline src/CommandHandler.js:74]: Remove console.log

File: src/commands/member_services/setvoicetopic.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    // Check if a user is in voicechannel
+    if (!message.member.voiceChannelID) {
+      await message.channel.send('You are not in a voice chat!').
+        then((msg) => {msg.delete(5000);});
+      return 'error';
+    }
+
+    const author = message.member;
+    // Check if a user has permission to change channel name
+    if (author.permissions.has('MANAGE_CHANNELS') ||
```
Comment (jerrylee17) [inline src/commands/member_services/setvoicetopic.js:33]: We could remove the permission requirement for normal members to use this command. @hsorathia @IsitaB @ThaiQ Thoughts?

File: src/commands/member_services/setvoicetopic.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        await message.channel.send('Successfully changed Name!')
+          .then((msg) => {msg.delete(5000);});
+      } catch (error) {
+        await message.channel.send('You are on cooldown!')
+          .then((msg) => {msg.delete(5000);});
+      } finally {
+        message.delete(5000);
+      }
+    }
+    return 'end async';
```
Comment (jerrylee17) [inline src/commands/member_services/setvoicetopic.js:55]: Does this do anything?

File: src/util/CooldownManager.js [hunk 7]
```diff
@@ -18,14 +18,14 @@ class CooldownManager {
 
   /**
    * Determine if a user needs to wait before invoking a command.
-   * @param {string} authorId The identification number of the author,
+   * @param {string} Id The identification number of the author,
```
Comment (jerrylee17) [inline src/util/CooldownManager.js:21]: Revert back to authorId

File: src/commands/member_services/setvoicetopic.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+    const author = message.member;
+    // Check if a user has permission to change channel name
+    if (author.permissions.has('MANAGE_CHANNELS') || 
+    author.permissions.has('ADMINISTRATOR')) {
+      const vcID = message.member.voiceChannelID;
+      const vc = author.guild.channels.get(vcID);
+      const oriname = vc.name;
+      // Function that edits the name
+      const fin = new Promise(function(resolve, reject){
```
Comment (jerrylee17) [inline src/commands/member_services/setvoicetopic.js:39]: Use an arrow function here
```
(resolve, reject) => { .... }
```

File: src/commands/member_services/setvoicetopic.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        vc.edit({name: str+'  : : ('+oriname+')'}).then(() => {resolve();});
+        setTimeout(function(){reject('timeout');}, 3000);
```
Comment (jerrylee17) [inline src/commands/member_services/setvoicetopic.js:41]: use arrow function here
```
setTimeout(()=>{reject('timeout');}, 3000);
```

File: package.json [hunk 10]
```diff
@@ -9,8 +9,15 @@
     "fix": "./node_modules/.bin/eslint ./ --fix"
   },
   "dependencies": {
+    "apollo-server": "^2.18.2",
```
Comment (jerrylee17) [inline package.json:12]: Don't need the following packages yet (Basically the ones you added):
```
apollo-server
apollo-server-express
express-graphql
graphql
graphql-compose
graphql-compose-mongoose
mongoose
```

Remove them and do an `npm install` to remove them from `package-lock.json`

**Review by jerrylee17 — CHANGES_REQUESTED** [review]
Really good code quality for first draft. Just a few minor changes from me.

File: src/commands/member_services/setvoicetopic.js [hunk 11]
```diff
@@ -0,0 +1,56 @@
+const Command = require('../Command');
+
+module.exports = new Command({
+  name: 'setvoicetopic',
+  description: 'Rename Voice Channel',
+  category: '',
```
Comment (jerrylee17) [inline src/commands/member_services/setvoicetopic.js:6]: Add a category. There's currently 
```
server management
mod
information
github
member services
```
I would put this under member services (if anyone can use this) at the moment.

File: src/commands/member_services/setvoicetopic.js (hunk 5 above)
Comment (jerrylee17) [reply src/commands/member_services/setvoicetopic.js:33]: If this permission were to stay here, the category would be under `mod`. If they were moved, the category would be `member services`.

**Review by jerrylee17 — CHANGES_REQUESTED** [review]
Set category to member services and remove the author permissions check

File: src/commands/member_services/setvoicetopic.js (hunk 5 above)
Comment (ThaiQ) [reply src/commands/member_services/setvoicetopic.js:33]: @jerrylee17 @Programmerryoki  I'm down to give member permission to create and rename their own study group channel (if that's what you meant @jerrylee17 )

**Review by ThaiQ — APPROVED** [review]

**Comment by ThaiQ** [comment]
I will just gonna approve this because I like this so far; @jerrylee17 will have the final saying

File: src/commands/member_services/setvoicetopic.js [hunk 12]
```diff
@@ -0,0 +1,62 @@
+const Command = require('../Command');
+
+module.exports = new Command({
+  name: 'setvoicetopic',
+  description: 'Rename Voice Channel, but can be only done 1 per 10 min',
```
Comment (IsitaB) [inline src/commands/member_services/setvoicetopic.js:5]: really?

Comment (Programmerryoki) [reply src/commands/member_services/setvoicetopic.js:5]: I mean 2 per 10 minutes...

**Review by IsitaB — APPROVED** [review]
Just fix the `1 per 10 min` comment and squash commits

File: src/commands/member_services/setvoicetopic.js (hunk 11 above)
Comment (jerrylee17) [reply src/commands/member_services/setvoicetopic.js:6]: Bump

File: src/commands/member_services/setvoicetopic.js (hunk 5 above)
Comment (jerrylee17) [reply src/commands/member_services/setvoicetopic.js:33]: Remove permission check

**Review by jerrylee17 — APPROVED** [review]

**Pull request merged**